marklogic-data-hub-central/src/test/java/com/marklogic/hub/central/controllers/steps/CustomStepControllerTest.java

```diff
@@ -0,0 +1,102 @@
+package com.marklogic.hub.central.controllers.steps;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.node.ArrayNode;
+
+import com.marklogic.hub.dataservices.StepService;
+
+import org.junit.jupiter.api.Test;
+import org.springframework.http.HttpStatus;
+import org.springframework.mock.web.MockHttpServletResponse;
+import org.springframework.security.access.AccessDeniedException;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.put;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+public class CustomStepControllerTest extends AbstractStepControllerTest {
+
+    private final static String PATH = "/api/steps/custom";
+    
+    /* used to create custom steps */
+    private final static String DEFAULT_JSON = "{"
+    		+ " \"name\": \"myCustomStep\" , "
+    		+ " \"description\": \"\" , "
+    		+ " \"selectedSource\": \"collection\" , "
+    		+ " \"sourceQuery\": \"cts.collectionQuery('test')\" , "
+    		+ " \"targetEntityType\": \"http://example.org/Customer-0.0.1/Customer\" , "
+    		+ " \"stepDefinitionName\": \"custom\" "
+    		+ "}";
+    private final static ObjectMapper MAPPER = new ObjectMapper();
+
+    public static class CustomStep {
+        public String name;
+        public String description;
+        public String selectedSource;
+        public String sourceQuery;
+        public String targetEntityType;
+        public String stepDefinitionName;
+    }
+
+    public static CustomStep newDefaultCustomStep(String name) {
+        CustomStep step = new CustomStep();
+        step.name = name;
+        step.description = "optional";
+        step.selectedSource = "collection";
+        step.sourceQuery = "cts.collectionQuery('test')";
+        step.targetEntityType = "http://example.org/Customer-0.0.1/Customer";
+        step.stepDefinitionName = "custom";
+        return step;
+    }
+
+    @Test
+    void userCanUpdateStep() throws Exception {
+        installOnlyReferenceModelEntities();
+        loginAsTestUserWithRoles("hub-central-custom-writer");
+
+        JsonNode new_node = MAPPER.readTree(DEFAULT_JSON);
+        StepService.on(getHubClient().getStagingClient()).saveStep("custom", new_node, false, true);

```

Comment:
Can you get the written custom step, assert for a few properties, then update those properties and verify if they all are updated ?

Reply:
Done.  The test is `userCanUpdateStep()`